app/js/components/manageApps/ManageApps.jsx

```diff
@@ -106,48 +108,61 @@ export default class ManageApps extends React.Component {
       let new_zip = new JSZip();
       new_zip.loadAsync(zipedAddon)
         .then((zip) => {
-          return zip.files['manifest.webapp'];
+          if (zip.files['manifest.webapp']) {
+            return zip.files['manifest.webapp'];
```

Comment:
Please refactor this code to be more like `zip.files['manifest.webapp'] ? zip.files['manifest.webapp'] : false`.
This is to avoid using un-necessary return statements.